src/components/ChatEntry.jsx

```diff
@@ -1,20 +1,35 @@
 import './ChatEntry.css';
+import TimeStamp from './TimeStamp';
+import PropTypes from 'prop-types';
+
+const ChatEntry = ({ id, sender, body, timeStamp, liked, onToggleLike }) => {
+  const handleLikeClick = () => {
+    onToggleLike(id);
+  };
```

Comment:
I like this pattern of sending just the `id` to `onToggleLike` since it keeps all the state management and message object creation confined to `App`.